#  Welcome to the Huber group

The Huber group develops statistical methods for modern biotechnologies, applies them to biological discovery, and translates them into reusable tools.

![Huber group hike on 2023-05-31](photos/2023-05-31-hike.jpg){width=640 fig-align="center"}

::: {.column-margin}
![Modern Statistics for Modern Biology textbook, with Susan Holmes: [online version](https://www.huber.embl.de/msmb). There is also a [print version published by CUP](https://www.cambridge.org/it/universitypress/subjects/statistics-probability/statistics-life-sciences-medicine-and-health/modern-statistics-modern-biology?format=HB).](photos/MSFMB-Cover2-compressed.jpg){width=240 fig-align="center"}

![Cellular neighborhood analysis of healthy and malignant lymph nodes based on single-cell resolution spatial proteomics by multiplexed immunohistochemistry.](photos/cellularneighborhoods.png){width=240 fig-align="center"}

![Cluster-free differential expression analysis of sc-RNA-seq data using LEMUR. [Paper link](https://doi.org/10.1038/s41592-023-01814-1).](photos/clusterfreeDE-2048x1649.png){width=240 fig-align="center"}

![Comparison of transformations for single-cell RNA-seq data. [Paper link](https://doi.org/10.1038/s41592-023-01814-1).](photos/Transformations-Fig1c.png)

![Ternary plots of relative sensitivities to targeted kinase inhibitors for a cohort of primary tumour samples of chronic lymphocytic leukaemia (CLL). [Paper link](https://doi.org/10.1172/JCI93801).](photos/JCI2018-fig_02_l.jpg)
:::


Our team studies biological systems by developing statistical and computational methods for the analysis of new data types and novel, large systematic datasets. These include single-cell and spatial omics, high-throughput drug- and CRISPR-based perturbation assays, and quantitative imaging. Projects range from applied data analysis for biological discovery to theoretical method development. We study fundamental biological model systems, as well as clinical samples for direct applications in biomedicine and precision oncology. We maintain an extensive network of collaborations. These include the 
[Molecular Medicine Partnership Unit] (MMPU) [Systems Medicine of Cancer Drugs](https://www.embl.org/about/info/mmpu/research-groups/systems-medicine-of-cancer-drugs), 
the ERC Synergy project [DECODE], 
the [ELLIS Unit Heidelberg], 
and the [Bioconductor] project.

Our interdisciplinary team comprises researchers from quantitative disciplines – such as physics, mathematics, statistics, computer science – and different fields of biology and medicine. We employ statistics and machine learning to discover patterns in data, understand mechanisms, and to build and investigate models. We pursue three main aims:

- Develop and improve new data generating technologies by powering them with the best statistical methods. This includes inference -- reasoning with uncertainty, making optimal decisions based on incomplete, noisy or overwhelming data -- as well as data exploration and visualization: helping scientists make and discoveries from large, complex datasets that they could not grasp otherwise.
- Make biological discoveries on drug-gene-environment interaction networks and their dynamical and context-dependent outcome in phenotypes. This includes the use of high-throughput perturbation data, single-cell, spatial, multimodal omics, and imaging.
- Make statistical methods more widely usable, not only for experts, but for the range of natural scientists. This aim is embodied by our engagement in open source, open science and the [Bioconductor] project.

{{< include _links.qmd >}}



