---
title: "Working with us"
---

### Current openings

- [Bioinformatician - Scientific Data Manager and Research Software Engineer - Precision Oncology and Multiomics](posts/2024-08-data-software-manager.html)
- [Postdoctoral position in statistical and AI method development for in-vivo Perturb-Seq in the ERC project DECODE](https://erc-decode.eu/)

::: {.column-margin}
[![](photos/EIPOD_LinC_RGB.png)](https://www.embl.org/about/info/postdoctoral-programme/eipod-linc-exploring-life-in-context/)

![](photos/DSC9178-2048x1401.jpeg){.lightbox}

![](photos/2021-10-21-retreathike.jpg){.lightbox}

![](photos/20240325_131603-sm.jpeg){.lightbox}
:::

Besides these specific calls, we are continually inviting applications for postdoc, PhD and internship positions. You can mix and match from two tracks:

- Method development in statistical methodology and computational biology
- Biological discovery through integrative data analysis

For Track 1, you will have strong quantitative and analytical skills, such as acquired through a degree in physics, statistics, mathematics, computer science or a related field. You have curiosity and motivation to work in interdisciplinary projects, which include generation of new data and their analysis, and are eager to get to grips with relevant areas of biology and the technologies used in biology research. You have experience in scientific computing and are familiar with one or several computer languages. Familiarity with R is definitively a plus.

For Track 2, you will have a training in life sciences and strong coding skills that enable you to undertake complex data transformations, integrative operations, applications of mathematical models and visualizations. You are driven by making fundamental discoveries by mining cutting-edge, large data sets.

To apply, please contact Wolfgang Huber with your CV, a brief statement of research interests, and examples of your work: besides your publications, this can include theses, research reports, talk slides, software projects (e.g. R packages, github projects) or data analysis reports (e.g., Quarto/Rmarkdown reports or Jupyter notebooks).

Please note that we recruit PhD candidates via the [EMBL international PhD programme](https://www.embl.org/about/info/embl-international-phd-programme/). You are highly welcome to discuss your application with Wolfgang Huber in advance.

## Partners for collaborative projects

The following list provides potential co-supervision partners for exciting opportunities to pursue interdisciplinary projects. The list is not exhaustive.

- [Davide Risso](https://drisso.github.io/), University of Padova: statistical method development for spatial omics, deep learning, latent space models
- [Anna Kreshuk](https://www.embl.org/groups/kreshuk/), EMBL: biological phenotyping in imaging-based latent spaces, foundation models, LLMs
- [Lars Steinmetz](https://www.embl.org/groups/steinmetz/), EMBL and  [Stanford](https://web.stanford.edu/group/steinmetzlab/cgi-bin/wordpress/): statistical method development for phenotyping environmental microbial communities using high-speed fluorescence image–enabled cell sorting
- [Michael Boutros](https://erc-decode.eu/), DKFZ: statistical method development for in-vivo Perturb-Seq and high-throughput imaging-based phenotyping
- [Sascha Dietrich](https://www.dietrichlab.de/), University Hospital Düsseldorf and [MMPU](https://www.embl.org/about/info/mmpu/research-groups/systems-medicine-of-cancer-drugs/): spatial omics, immunotherapies and the tumor microenvironment, immunooncology
- [Thorsten Zenz](https://www.lymphomaresearchzurich.com/), University Hospital Zürich: 'omics of drug sensitivity in blood cancers
- [Nassos Typas](https://www.embl.org/groups/typas/), EMBL: high-throughput phenotyping of microbial communities under genetic and chemical perturbations; antibiotica, phages
- [Misha Savitski](https://www.embl.org/groups/savitski/), EMBL: stability proteomics
- [Sinem Saka](https://www.embl.org/groups/saka/), EMBL: spatial biology from molecules to tissues
- [Oliver Stegle](https://www.embl.org/groups/stegle/), EMBL: statistical genomics and systems genetics
- [Victoria Ingham](https://www.klinikum.uni-heidelberg.de/zentrum-fuer-infektiologie/parasitology-unit/research/ingham-lab), University Hospital Heidelberg: Parasitology, insecticide resistance and malaria
- [Aedín Culhane](https://www.ul.ie/hri/prof-aedin-culhane/), University of Limerick: dimension reduction and latent spaces for (single cell resolution) omics data 
- [Nikos Ignatiadis](https://nignatiadis.github.io/), Chicago: statistical methodology
- **Susan Holmes**, Stanford: statistical methodology
- **The Bioconductor Community**


